### ISSUE TITLE
browserify attempts to load all language files

### ISSUE THREAD

**Issue by clayzermk1** [foretagsplatsen/numbro#79]
The patch in #66 automatically loads all language files for any commonjs scenario, [browserify](https://github.com/substack/node-browserify) included.

While I can't really complain about doing this on the server with Node.js, doing this in the browser adds unnecessary bloat that I can't easily remove.

Possible solutions off the top of my head:
* provide a way to specify an array of languages which defaults to all languages on Node
* create special logic for Node within the commonjs scope

As a side effect, this might help with #70.

**Comment by clayzermk1** [comment]
Reverting #66 fixes this issue along with #70.

**Comment by BenjaminVanRyseghem** [comment]
But it also reverse the fix for #62 :confused:

**Comment by clayzermk1** [comment]
Perhaps there is a way of implementing #62 without putting it in the module definition? Can we make a loader function that then uses the appropriate loader for the framework?

**Comment by clayzermk1** [comment]
Submitted a PR for a trivial patch to check `process.title === 'node'`.

I know browserify uses "browser" and I don't know for certain about WebPack but I don't think it uses "node".

I don't think this is an ideal solution to the overall problem - it just punts it to another day but gets browserify folks back up and running without breaking #62.

**Comment by clayzermk1** [comment]
@BenjaminVanRyseghem have you had the chance to give this topic any further thought? I'd like to get a solution going that goes in the right direction for everyone and the project in general.

**Comment by BenjaminVanRyseghem** [comment]
the `process.title === 'node'` solution is not the most elegant one, but it seems to work.

It's better to have a half ugly working solution than a theoretical perfect not working solution :smile:

Then maybe the *browserify* users have more input on this?

**Comment by clayzermk1** [comment]
Agreed, like I said, it's a punt.

The issue with the module definition is that Node.js specific code got placed in a CommonJS section. This would theoretically break all CommonJS module loaders that aren't Node.js. Browserify (#70) and WebPack (#74) (CommonJS implementers) are clearly broken because of this.

I feel like I'm taking this issue off topic though. My point of this issue was to address all languages modules being loaded when using a CommonJS implementation. That should not be done on the client, and in my opinion it should not be done in the general case. I think a language loading function is a superior solution. It's hardly troublesome to have users specify which locales they would like to load in addition to the default en_XX locale.

I envision something like the following:
```javascript
var numbro = require('numbro').includeLocales([ 'jp_JP', 'de_DE' ]); // loads just Japanese and German locales
var numbro = require('numbro').includeAllLocales(); // loads all locales, can be an alias to includeLocales()
```

**Comment by BenjaminVanRyseghem** [comment]
yep, it looks like this whole part was a hack to have it out, but it feels like it's time to have a look at it now :smile:

your API proposal sounds good to me :smile: , if you want to propose a PR going in this direction, that will be awesome :smile:

**Comment by clayzermk1** [comment]
I'll see what I can do, I have a feeling it'll take some time to fix all the tests etc.

Is anyone else working towards a solution for this?

**Comment by BenjaminVanRyseghem** [comment]
I can help if you want, but I have really little time available those days

(I created a Gitter room for numbro if you want to discuss there)

**Comment by clayzermk1** [comment]
#71 seems to related as well.

**Comment by BenjaminVanRyseghem** [comment]
can you check that #101 fixes this issue?

**Comment by clayzermk1** [comment]
I can confirm that this issue is resolved by #101.
